controllers/spiaccesstoken_controller.go

```diff
@@ -195,7 +197,14 @@ func (r *SPIAccessTokenReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	// persist the SP-specific state so that it is available as soon as the token flips to the ready state.
 	sp, err := r.ServiceProviderFactory.FromRepoUrl(ctx, at.Spec.ServiceProviderUrl, req.Namespace)
 	if err != nil {
-		if uerr := r.flipToExceptionalPhase(ctx, &at, api.SPIAccessTokenPhaseError, api.SPIAccessTokenErrorReasonUnknownServiceProvider, err); uerr != nil {
+		var reason api.SPIAccessTokenErrorReason
+		var validationErr validator.ValidationErrors
+		if stderrors.As(err, &validationErr) {
+			reason = api.SPIAccessTokenErrorUnsupportedServiceProviderConfiguration
+		} else {
```

Comment:
Is it necessary here (and in other places) to have the new ErrorReason? I am not opposed, but the number of different reasons keeps rising and here it makes the code more "complex", and I think it might be enough to just keep it as `SPIAccessTokenErrorReasonUnknownServiceProvider` and explain more context in the ErrorMessage. WDYT?

Reply:
Good question. I understand your point. At the same time, it we will need to find CR-s to fix or count them etc. 
separate status is better imo.